pkgs/development/python-modules/tensorflow/default.nix

```diff
@@ -53,7 +53,16 @@ let
     if cudaSupport then cudaPackages.backendStdenv
     else if originalStdenv.isDarwin then llvmPackages_11.stdenv
     else originalStdenv;
-  inherit (cudaPackages) cudatoolkit cudnn nccl;
+  inherit (cudaPackages) cudatoolkit nccl;
+  # use compatible cuDNN (https://www.tensorflow.org/install/source#gpu)
+  cudnn = cudaPackages.cudnn_8_6;
```

Comment:
nit, not necessarily a blocker: i think it'd preferable to do something like
```
, cudnn ? cudaPackages.cudnn_8_6
```
in the arguments list instead. This would allow end-users to override if they so desire.

Also have we tested against `cudaPackages.cudnn`? What does that failure look like? Let's list the failure logs in a code comment if we can.

Reply:
cudaPackages.cudnn triggered https://github.com/tensorflow/tensorflow/issues/60398 . I'll put in a comment.

Reply:
> This would allow end-users to override if they so desire.

I opened a separate PR for this (https://github.com/NixOS/nixpkgs/pull/264710) after verifying it works as intended.